src/components/HeaderWithLanguage/EmbedInformation/EmbedConnection.tsx

```diff
@@ -46,11 +46,20 @@ const EmbedConnection = ({ t, id, type }: Props & tType) => {
   const [concepts, setConcepts] = useState<SearchConceptType[]>();
 
   useEffect(() => {
+    let shouldUpdateState = true;
     if (id) {
-      searchArticles(searchObjects(id, type)).then(result => setArticles(result.results));
+      searchArticles(searchObjects(id, type)).then(result => {
+        if (shouldUpdateState) setArticles(result.results);
+      });
       type === 'image' &&
-        searchConcepts(searchObjects(id, type)).then(result => setConcepts(result.results));
+        searchConcepts(searchObjects(id, type)).then(result => {
+          if (shouldUpdateState) setConcepts(result.results);
+        });
     }
+
+    return () => {
+      shouldUpdateState = false;
+    };
```

Comment:
Siden du definerer `shouldUpdateState` inn i useEffecten, er jeg litt usikker på om denne cleanUp funksjonen gjør noe. Uansett om den settes til false eller ikke vil verdien bli true når useEffecten kjøres, eller tenker jeg feil?

Reply:
Du tenker feil. Den settes til false i det komponenten unmountes, som vil si at om den har blitt unmounted (dvs før promisene har rukket å komme til `.then`) så vil vi ikke oppdatere staten for da blir react sinte og gir oss feilmeldinger.
Det er vel egentlig bare en noop, men siden det _KAN_ bety minnelekkasje (ikke i dette tilfellet egt) så blir react sint og gir oss røde meldinger i konsollen.